### PR TITLE
FLPATH-3438 Enable source CRUD tests, add Kafka config, add Sources tests, update IQE profiles

### PR DESCRIPTION
**Pull request by testetson22**
This updates IQE test infrastructure to enable newly-fixed tests and improve containerized test execution:

#### 1. Enable Source CRUD Tests

**Background:** FLPATH-3423 (backend PATCH 500 error) has been fixed in Koku.

**Changes:**
- Set `SKIP_SOURCE_CRUD_TESTS=false` in `iqe-filters.sh`
- Update smoke profile filter to include `test_api_ocp_source_crud`
- Update documentation to mark FLPATH-3423 as resolved

#### 2. Kafka Configuration for IQE Container

**Problem:** `test_api_ocp_ros_kafka_content` requires Kafka broker configuration, which wasn't available in containerized runs.

**Fix:** In `run-iqe-tests.sh`:
- Extract Kafka bootstrap server from Strimzi CR
- Pass `DYNACONF_BROKER__hostname`, `DYNACONF_BROKER__port`, `DYNACONF_BROKER__securityProtocol` to IQE pod
- Enables ROS Kafka test to run in CI

#### 3. Update Full IQE Profile

**Problem:** 90-day date range tests fail with HTTP 400 due to `RETAIN_NUM_MONTHS=3` in the chart.

**Change:** Skip these tests in the `full` profile until COST-7253/COST-573 are completed:
- Set `SKIP_DATE_RANGE_TESTS=true` for `full` profile
- Update test count estimate (~3200 tests, ~90 min)
- Add comments referencing FLPATH-4131, COST-7253, COST-573

#### 4. JUnit Suite Name

Add `-o junit_suite_name=iqe-cost-management-onprem` to pytest commands to distinguish IQE results from chart test results in CI reports.

#### 5. New Chart-Level Sources Tests

Added to `tests/suites/sources/test_sources_api.py`:
- `TestSourcesPauseResume` - Verifies PATCH pause/resume works (FLPATH-3486)
- `TestSourcesTimestamps` - Verifies `updated_timestamp` is modified on PATCH (FLPATH-3420)

#### 6. IQE Loop Script

Added `scripts/qe/run-iqe-loop.sh` for local testing:
- Runs tests multiple times with cleanup between iterations
- Helps identify flaky tests
- Provides summary table of results

### Testing

- Verified source CRUD tests pass in CI
- Verified Kafka configuration enables ROS test in container
- Full profile IQE run: **2260 passed**, 32 failed (unrelated volume edge cases)
- Chart tests pass including new sources tests

### Related Issues

- FLPATH-3438 (On-Prem IQE Test Enablement)
- FLPATH-3423 (Source CRUD backend fix - resolved)
- FLPATH-3486 (Source pause/resume)
- FLPATH-3420 (Source updated_timestamp)
- FLPATH-4131 (90-day test enablement - blocked on COST-7253/COST-573)

Marked as Draft until IQE MR is merged with required source CRUD test fixes

### CODE REVIEW

File: scripts/qe/run-iqe-loop.sh [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    route=$(kubectl get route -n "$NAMESPACE" cost-onprem-api -o jsonpath='{.spec.host}' 2>/dev/null) || \
+    route=$(kubectl get route -n "$NAMESPACE" cost-onprem-gateway -o jsonpath='{.spec.host}' 2>/dev/null) || \
+    route=$(kubectl get route -n "$NAMESPACE" -l app.kubernetes.io/component=ingress -o jsonpath='{.items[0].spec.host}' 2>/dev/null)
+    
+    if [[ -z "$route" ]]; then
+        log_error "Could not find API gateway route"
+        return 1
+    fi
+    echo "https://${route}"
+}
```
Comment (chadcrum) [inline scripts/qe/run-iqe-loop.sh:118]: The cleanup filter in `list_test_sources()` matches `test_cost_*` prefixed sources, but the new tests in `test_sources_api.py` create sources named `pause-test-*`, `ts-test-*`, `ts-patch-*`. If someone uses the loop script to run these source tests, the cleanup won't catch the orphaned sources — which is exactly the scenario the loop script is built for. Might want to broaden the filter or standardize the naming.

**Review by chadcrum — APPROVED** [review]
Looks good — solid cleanup patterns in the tests, Kafka fallback is nice, and the loop script fills a real gap. One minor thing on the cleanup filter.

**Review by masayag — APPROVED** [review]
